### PR TITLE
[CI] Remove Doxygen docs from intel.github.io/llvm-docs/

### PR DESCRIPTION
**Pull request by uditagarwal97**
Prequel of https://github.com/intel/llvm/pull/15483

This PR removes Doxygen docs/links from intel.github.io/llvm-docs/. This is done to reduce the size of the website so that it can be hosted via artifacts in #15483. [intel/llvm-docs](https://github.com/intel/llvm-docs) repository will soon be decommissioned.

### CODE REVIEW

File: .github/workflows/sycl-docs.yml [hunk 1]
```diff
@@ -52,9 +50,7 @@ jobs:
         git rm -rf .
         touch .nojekyll
         yes | \cp -rf $GITHUB_WORKSPACE/build/tools/sycl/doc/html/* .
-        mv $GITHUB_WORKSPACE/build/tools/sycl/doc/doxygen/html doxygen/
         mv $GITHUB_WORKSPACE/build/tools/clang/docs/html clang/
-        mv $GITHUB_WORKSPACE/build/tools/clang/docs/doxygen/html clang_doxygen/
         git config --global user.name "iclsrc"
         git config --global user.email "ia.compiler.tools.git@intel.com"
         git add .
```
Comment (bader) [inline .github/workflows/sycl-docs.yml:56]: @uditagarwal97, could you please remove `-f` from `git push -f` command below?

Comment (uditagarwal97) [reply .github/workflows/sycl-docs.yml:56]: Done

File: .github/workflows/sycl-docs.yml [hunk 2]
```diff
@@ -25,16 +25,14 @@ jobs:
...
       run: |
-        sudo apt-get install -y doxygen graphviz ssh ninja-build libhwloc-dev
+        sudo apt-get install -y graphviz ssh ninja-build libhwloc-dev
         sudo pip3 install -r repo/llvm/docs/requirements.txt
     - name: Build Docs
       run: |
         mkdir -p $GITHUB_WORKSPACE/build
         cd $GITHUB_WORKSPACE/build
         python $GITHUB_WORKSPACE/repo/buildbot/configure.py -w $GITHUB_WORKSPACE \
         -s $GITHUB_WORKSPACE/repo -o $GITHUB_WORKSPACE/build -t Release --docs
```
Comment (bader) [inline .github/workflows/sycl-docs.yml:35]: Potentially it can be useful to verify that developers can build Doxygen documentation locally if needed. Currently, we build Doxygen documentation during pre-commit testing w/o deploying the documentation.

I don't have a strong opinion about keeping Doxygen documentation buildable. IMHO, even if broken, it should be easy to fix.
If any of code owners have an opinion, feel free to leave a comment.

File: .github/workflows/sycl-docs.yml [hunk 3]
```diff
@@ -52,11 +50,9 @@ jobs:
...
         touch .nojekyll
         yes | \cp -rf $GITHUB_WORKSPACE/build/tools/sycl/doc/html/* .
-        mv $GITHUB_WORKSPACE/build/tools/sycl/doc/doxygen/html doxygen/
         mv $GITHUB_WORKSPACE/build/tools/clang/docs/html clang/
-        mv $GITHUB_WORKSPACE/build/tools/clang/docs/doxygen/html clang_doxygen/
         git config --global user.name "iclsrc"
         git config --global user.email "ia.compiler.tools.git@intel.com"
         git add .
         git diff-index --quiet HEAD || git commit --amend -m "Update docs" -s
-        git push -f
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-docs.yml:62]: I'd prefer this line to be a separate PR

**Review by aelovikov-intel — APPROVED** [review]

File: .github/workflows/sycl-docs.yml [hunk 4]
```diff
@@ -25,16 +25,14 @@ jobs:
         path: repo
     - name: Install deps
       run: |
-        sudo apt-get install -y doxygen graphviz ssh ninja-build libhwloc-dev
+        sudo apt-get install -y graphviz ssh ninja-build libhwloc-dev
```
Comment (bader) [inline .github/workflows/sycl-docs.yml:28]:
```suggestion
        sudo apt-get install -y ssh ninja-build libhwloc-dev
```
Do we need graphviz after Doxygen removal?

**Review by bader — APPROVED** [review]

File: .github/workflows/sycl-docs.yml (hunk 4 above)
Comment (uditagarwal97) [reply .github/workflows/sycl-docs.yml:28]: Yes, we still need graphviz. Otherwise I get the following error (https://github.com/uditagarwal97/llvm/actions/runs/11018898111/job/30600198171?pr=23#step:4:996):

```
cd /home/runner/work/llvm/llvm/build/tools/clang/docs && /usr/local/bin/cmake -E env /usr/local/bin/sphinx-build -b html -d /home/runner/work/llvm/llvm/build/tools/clang/docs/_doctrees-clang-html -q -t builder-html -D version=20 -D release=20.0.0git -tPreRelease -W /home/runner/work/llvm/llvm/build/tools/clang/docs /home/runner/work/llvm/llvm/build/tools/clang/docs/html

Warning, treated as error:
dot command 'dot' cannot be run (needed for graphviz output), check the graphviz_dot setting
```

**Comment by uditagarwal97** [comment]
@intel/llvm-gatekeepers the PR is ready for merge. E2E test failures are unrelated and also reported in other PRs (like https://github.com/intel/llvm/actions/runs/11018636840/job/30604700124)

**Comment by elliottbinder** [comment]
> esimd lgtm

I've been regularly referencing the [DPC++ Explicit SIMD API](https://intel.github.io/llvm-docs/doxygen/group__sycl__esimd.html) for my development work. Will this no longer be hosted anywhere?

**Comment by sarnex** [comment]
@uditagarwal97 @bader Can you guys please address Elliot's question? It looks like he was using the Doxygen.

**Comment by bader** [comment]
@elliottbinder, could you provide more details about your use case? The Doxygen documentation intended to be used by DPC++ compiler developers, so we haven't considered other scenarios.

**Comment by elliottbinder** [comment]
I work in HPC research and appreciate any access to whatever features are available in the ISA of a CPU, GPU, or whatever processor. I'm leveraging the matrix acceleration instructions (DPAS/W) present on Intel GPUs for one project, and I've been using the ESIMD API to produce instructions that I want. This gives me access to instructions and functionality I'm not sure I'd be able to use otherwise (horizontal reduction, using different vector lengths and operations, block memory operations, register region addressing, cache hints). If you know of a different way to get this functionality without ESIMD, I'd be interested to know.

[Intel's Intrinsics Guide](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html) has been a reliable resource for CPUs for many years. If you take away this documentation for your GPUs, as far as I know how, I'd be stuck programming the GPU using a SIMT interface and don't get to leverage all the useful functionality that's already sitting there in silicon. In my experience, relying solely on compilers often leads to inefficient code, or having to write your code in a very specific way that may change if the underlying compiler heuristics change. It is simpler for folks in my line of work to just produce the instruction sequences we'd like directly.

Do you know how the LinAlg/FFT/ML library teams are programming the GPUs? I'd expect to see similar approaches there as well.

**Comment by asudarsa** [comment]
Hi @elliottbinder 
Can you please check if the documentation here is sufficient for your purpose?

https://intel.github.io/llvm/

Thanks

**Comment by bader** [comment]
@asudarsa, based on the link @elliottbinder provided above, he uses Doxygen documentation generated from SYCL headers. That part has been removed from the hosted rendered version (it still can be generated locally though).

We don't use Doxygen generated documentation for user faced API. This API is supposed to be documented in the extension specification. @elliottbinder, https://github.com/intel/llvm/tree/sycl/sycl/doc/extensions/supported/sycl_ext_intel_esimd should work for your use case.

The Doxygen is built from sources, ESIMD headers can be used as well, although they are less readable than Doxygen generated webpages.
